.github/workflows/release.yml

```diff
@@ -0,0 +1,49 @@
+name: Create Zipapp and Release
```

Comment:
Should we do this in CircleCI since that's where the rest of the CI for this repo already runs?

Reply:
I'm not that familiar with CircleCi so decided to do it in GHA instead. I did briefly take a look at it but GHA has a nicer and easier way of implementing this as part of their built-in action and uses pre-defined credentials. If this is a biggie, I can take a look at it but if you're happy to roll with this and I can take a proper look at this when I get some more time.